storage/range_command.go

```diff
@@ -344,11 +344,20 @@ func (r *Range) InternalRangeLookup(batch engine.Engine, args *proto.InternalRan
 	// We want to search for the metadata key just greater than args.Key. Scan
 	// for both the requested key and the keys immediately afterwards, up to
 	// MaxRanges.
-	metaPrefix := proto.Key(args.Key[:len(engine.KeyMeta1Prefix)])
-	nextKey := proto.Key(args.Key).Next()
+	var metaPrefix proto.Key
+	var startKey proto.Key
+	if len(args.Key) > 0 {
```

Comment:
`KeyMeta1Prefix + KeyMin` is already defined: it's the meta1 key corresponding to the first meta2 key. We'd need to change the encoding so that meta1 keys include the meta2 prefix instead of stripping it off, so this is unambiguous. I started pulling on that thread but it was non-trivial.

I've extracted the DecodeRangeMetaKey function.